ubuntu/ubuntu.go

```diff
@@ -78,11 +78,14 @@ func Update() error {
 	dir := filepath.Join(utils.CacheDir(), cveTrackerDir)
 	for _, url := range repoURLs {
 		_, err = gc.CloneOrPull(url, dir, "master", false)
-		if err != nil {
-			log.Printf("failed to clone or pull: %s: %v", url, err)
-			continue
+		if err == nil {
+			break
+		}
+		log.Printf("failed to clone or pull: %s: %v", url, err)
+		log.Printf("removing %s directory", cveTrackerDir)
+		if err := os.RemoveAll(dir); err != nil {
```

Comment:
Even if the directory doesn't exist, doesn't it return an error? I'm concerned that the first try fails immediately and it didn't create a directory. Then, `os.RemoveAll` fails and the second try will not be performed.

Reply:
actually that makes sense, I'll update the PR.

Reply:
```
// it encounters. If the path does not exist, RemoveAll
// returns nil (no error).
```
actually from [os.RemoveAll docs](https://github.com/golang/go/blob/d0d38f0f707e69965a5f5a637fa568c646899d39/src/os/path.go#L63-L64), so if dir does not exist, this will not matter.

Reply:
Thank you for confirming.